### ISSUE TITLE
Feature Request:  Cosmetic Transition Effects

### ISSUE THREAD

**Issue by ChristianSilvermoon** [Wohlstand/TheXTech#63]
There are a variety of transition effects that do not exist in (either SMBX or) TheXTech

Namely
• Screen fading to and from black on certain warps
• Screen fading to and from black on return to World Map
• Zoomy Bubble Effect on entering a level from World Map (EX: SMB3 from All Stars)
• SMW Goal Tape Exit (This is actually what prompted this, as the sound playing, but no zoomy bubble thing has always been slightly jarring to me)
• The Key Hole Effect (could zoom the keyhole graphic and shrink it to accomplish the effect with no additional assets maybe?)

Doors and pipes should not do a fade transition if the exit is already on screen, ideally.

It may be desirable, if a pipe exit is in the same section to pan over to it, but this could result in certain levels showing off things that weren't mean to be seen if this occurred... so maybe something enable/disable-able per level/episode?

In order to leave gameplay **unaffected** by the addition of the transitions (IE levels with time sensitive events, etc.) the game should probably enter a paused state during transitions.

For Multiplayer, transitions would probably need to be disabled.

This would mostly just be cosmetic polish, probably disabled for speedruns?

**Comment by Wohlstand** [comment]
I do think to make that in some cases, but not now yet, probably once upon, but I really like the feature! :fox_face: :+1:  I made some of that at PGE Engine already! :fox_face: 

> This would mostly just be cosmetic polish, probably disabled for speedruns?

I the condition affects the final time count, they will be disabled. At least I'll try to make these effects to be harmless for cycle counts, just perform them in parallel with the cycle counting. :thinking:

**Comment by Wohlstand** [comment]
> The Key Hole Effect (could zoom the keyhole graphic and shrink it to accomplish the effect with no additional assets maybe?)

This one will need me to make the way to draw polygon by SDL2 Renderer API, and also I need the way to better cut around the player's sprite while the keyhole gets shrunk down.

**Comment by Wohlstand** [comment]
Everything except the keyhole has been implemented!

I guess for the keyhole I'll make the new issue, that's a thing which is a tricky one, will put for 1.3.6+ plan. This issue I'll close.

**Comment by Wohlstand** [comment]
To continue discussion of keyhole, I made this: #237
